.actions/assistant.py

```diff
@@ -130,107 +143,17 @@ def replace_oldest_ver(requirement_fnames: Sequence[str] = REQUIREMENT_FILES_ALL
-
-    @staticmethod
-    def _find_pkgs(folder: str, pkg_pattern: str = "lightning") -> List[str]:
-        """Find all python packages with spec.
-
-        pattern in given folder, in case `src` exists dive there.
-        """
-        pkg_dirs = [d for d in glob.glob(os.path.join(folder, "*")) if os.path.isdir(d)]
-        if "src" in [os.path.basename(p) for p in pkg_dirs]:
-            return AssistantCLI._find_pkgs(os.path.join(folder, "src"), pkg_pattern)
-        pkg_dirs = list(filter(lambda p: pkg_pattern in os.path.basename(p), pkg_dirs))
-        return pkg_dirs
-
-    @staticmethod
-    def mirror_pkg2source(pypi_folder: str, src_folder: str) -> None:
-        """From extracted sdist packages overwrite the python package with given pkg pattern."""
-        pypi_dirs = [d for d in glob.glob(os.path.join(pypi_folder, "*")) if os.path.isdir(d)]
-        for pkg_dir in pypi_dirs:
-            for py_dir in AssistantCLI._find_pkgs(pkg_dir):
-                dir_name = os.path.basename(py_dir)
-                py_dir2 = os.path.join(src_folder, dir_name)
-                shutil.rmtree(py_dir2, ignore_errors=True)
-                shutil.copytree(py_dir, py_dir2)
-
     @staticmethod
     def copy_replace_imports(
         source_dir: str, source_import: str, target_import: str, target_dir: Optional[str] = None
     ) -> None:
         """Recursively replace imports in given folder."""
-
         source_imports = source_import.strip().split(",")
         target_imports = target_import.strip().split(",")
-        assert len(source_imports) == len(target_imports), (
-            "source and target imports must have the same length, "
-            f"source: {len(source_import)}, target: {len(target_import)}"
-        )
-
-        if target_dir is None:
-            target_dir = source_dir
-
-        ls = _retrieve_files(source_dir)
-
-        for fp in ls:
-            if fp.endswith(".py"):
-                with open(fp, encoding="utf-8") as fo:
-                    py = fo.readlines()
-
-                for source_import, target_import in zip(source_imports, target_imports):
-                    for i, ln in enumerate(py):
-                        py[i] = re.sub(rf"([^_]|^){source_import}([^_\w]|$)", rf"\1{target_import}\2", ln)
-
-                if target_dir:
-                    fp_new = fp.replace(source_dir, target_dir)
-                    os.makedirs(os.path.dirname(fp_new), exist_ok=True)
-                else:
-                    fp_new = fp
-
-                with open(fp_new, "w", encoding="utf-8") as fo:
-                    fo.writelines(py)
-            elif not fp.endswith(".pyc"):
-                fp_new = fp.replace(source_dir, target_dir)
-                os.makedirs(os.path.dirname(fp_new), exist_ok=True)
-                if os.path.abspath(fp) != os.path.abspath(fp_new):
-                    shutil.copy2(fp, fp_new)
+        copy_replace_imports(source_dir, source_imports, target_imports, target_dir=target_dir)
```

Comment:
Does it call itself? I guess not but then lest call the function differently

Reply:
One is the assistant staticmethod, the other the function. They do the same so they should be called equally